### PR TITLE
test: migrate AccordianReferences tests from Enzyme to React Testing Library

### PR DESCRIPTION
**Pull request by vishvamsinh28**
Rewrote AccordianReferences test suite using React Testing Library. Improved clarity, ensured full coverage including edge cases.

### CODE REVIEW

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianReferences.test.js [hunk 1]
```diff
@@ -65,47 +73,62 @@ describe('<AccordianReferences>', () => {
...
 
   beforeEach(() => {
-    wrapper = shallow(<AccordianReferences {...props} />);
+    jest.clearAllMocks();
   });
 
-  it('renders without exploding', () => {
-    expect(wrapper).toBeDefined();
-    expect(wrapper.exists()).toBe(true);
+  it('renders the component structure correctly without crashing', () => {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianReferences.test.js:79]: this is unnecessary title change that confuses the diffs. Why make it? "without exploding" was sufficiently explanatory
